### ISSUE TITLE
Error Estimating Gene Set Expression Scores

### ISSUE THREAD

**Issue by nadineparker** [douglasyao/mesc#5]
I have tried to get gene set expression scores using eQTL summary statistics and I get this error:

File "./../../mesc/run_mesc.py", line 244, in <module>
    ss.get_expression_scores(args)
  File "/path/to/mesc/expscore_sumstat.py", line 269, in get_expression_scores
    gene_gset_dict[gene].append('{}_Cis_herit_bin_{}'.format(k, gset_gene_bins[ix]))
IndexError: index 30 is out of bounds for axis 0 with size 30

At first the code gave me this error on chromosome 1 after the 5th gene set so as a test I reran the code with only the first 4 gene sets. This allowed chromosome 1 to complete but then the same error occurred for chromosome 2. It's not clear to me what the issue is based on the code in expscore_sumstat.py.

**Comment by douglasyao** [comment]
Hi Nadine,

Can you send me a snippet of the input data that produces this error? It
may be a bug. Thanks!

-Doug

**Comment by nadineparker** [comment]
Hi,
attached are 9 genesets I've been using. So far the first 2 complete the expression score calculation for all chromosomes when run individually (single gene set at a time) but the third halts at chromosome 2. Still attempting individual runs of the other gene sets.
[mesc_share.txt](https://github.com/douglasyao/mesc/files/8933553/mesc_share.txt)

**Comment by douglasyao** [comment]
Hi,

Sorry, can you send me the actual input files (including the eQTL summary
statistics and input gene set files)? If they are too large, can you send
me a subset of the data that still produces the error? Thanks!

-Doug

**Comment by nadineparker** [comment]
Hi,
Is it possible to send the eQTL data another way? The file is large and it's difficult to tell what section to send since I'm not sure where the problem arises.
Of note, I have successfully run the overall eQTL expression scores with the same eQTL data previously (implementing the code changes mentioned in issue #4). The problem seems to arise only for the gene set scores.

**Comment by douglasyao** [comment]
Hi,

Unfortunately it's hard for me to pinpoint the bug without a
reproducible example. Are you running the latest version of MESC? For some
people, updating to the latest version resolved the issues they had with
the software. If the issue persists, can you send me the log file from
running the method?

Thanks, Doug

**Comment by nadineparker** [comment]
Hi Douglas,
I was able to get a subset of the eQTL data to produce a similar error. Attached are the gene sets and subset of eQTL data. 

As for version of MESC, I downloaded the latest version 1 month ago. In that version I had a similar issue as seen in issue #1 except at the end of the expscore_sumstat.py script from line 357 onwards. The changes mentioned in issue #4 corrected the problem and I was able to run mesc to calculate overall eQTL expression scores and to calculate mediated heritability in different traits. 

Thanks
[test_subset_eQTL_2.1.txt.tar.gz](https://github.com/douglasyao/mesc/files/8960719/test_subset_eQTL_2.1.txt.tar.gz)
[GeneSets.txt](https://github.com/douglasyao/mesc/files/8960722/GeneSets.txt)
